### PR TITLE
implementing configMaps

### PR DESCRIPTION
**Pull request by BrentDorsey**
* adding config files in a central location to facilitate creating kubernetes configMaps for the following pods
 * removing config files from application folders
    * refactoring  to mount all configuration files because configMap's don't support mounting only specific files within a directory
    * airflow-config
       * commenting r installs due to unauthenticated error
    * cassandra-config
    * hdfs-config
    * jupyterhub-config
    * redis-config
        * removing REDIS_HOME environment variable due to data and config mounts failing
    * spark-config
    * spark yaml's implementing worker and master role and version
    * zeppelin-config
implementing hpa - horizontal pod autoscalers

removed duplicate tensorflow install from airflow and zeppelin dockerfile which are both built from package-spark-2.1.0, tensorflow is installed [here in package-spark-2.1.0](https://github.com/fluxcapacitor/pipeline/blob/master/package.ml/spark/2.1.0/Dockerfile#L84)

### CODE REVIEW

**Comment by BrentDorsey** [comment]
To create configMaps simply run the shell scripts in [pipeline/kubernetes.ml/config-maps/](https://github.com/fluxcapacitor/pipeline/tree/b11f17f062329b1ea1714cc0af737c580a1ccfe1/kubernetes.ml/config-maps)

File included in each configMap:

**cassandra-config**
```
kubectl delete configmap cassandra-config
kubectl create configmap cassandra-config --from-file=cassandra
```

- cassandra-env.ps1
- cassandra-env.sh
- cassandra-env.sh.orig
- cassandra-jaas.config
- cassandra-rackdc.properties
- cassandra-topology.properties
- cassandra.yaml
- cassandra.yaml.orig
- commitlog_archiving.properties
- cqlshrc.sample
- hotspot_compiler
- jvm.options
- logback-tools.xml
- logback.xml
- metrics-reporter-config-sample.yaml
- README.txt

**hdfs-config**
```
kubectl delete configmap hdfs-config
kubectl create configmap hdfs-config --from-file=hdfs
```

- capacity-scheduler.xml
- configuration.xsl
- container-executor.cfg
- core-site.xml
- core-site.xml.orig
- hadoop-env.cmd
- hadoop-env.sh
- hadoop-metrics2.properties
- hadoop-metrics.properties
- hadoop-policy.xml
- hdfs-site.xml
- hdfs-site.xml.orig
- httpfs-env.sh
- httpfs-log4j.properties
- httpfs-signature.secret
- httpfs-site.xml
- kms-acls.xml
- kms-env.sh
- kms-log4j.properties
- kms-site.xml
- log4j.properties
- mapred-env.cmd
- mapred-env.sh
- mapred-queues.xml.template
- mapred-site.xml.template
- slaves
- ssl-client.xml.example
- ssl-server.xml.example
- yarn-env.cmd
- yarn-env.sh
- yarn-site.xml

**spark-config**
```
kubectl delete configmap spark-config
kubectl create configmap spark-config --from-file=spark
```

- docker.properties.template
- fairscheduler.xml
- fairscheduler.xml.template
- hive-site.xml
- log4j.properties.template
- metrics.properties.template
- slaves.template
- spark-defaults.conf
- spark-defaults.conf.template
- spark-env.sh.template

**redis-config**
```
kubectl delete configmap redis-config
kubectl create configmap redis-config --from-file=redis
```

- redis.conf

**presto-catalog**
```
kubectl delete configmap presto-catalog
kubectl create configmap presto-catalog --from-file=presto
```

- cassandra.properties
- hive.properties
- jmx.properties
- kafka.properties
- mysql.properties
- redis.properties

**jupyterhub-config**
```
kubectl delete configmap jupyterhub-config
kubectl create configmap jupyterhub-config --from-file=jupyterhub
```

- jupyterhub_config.py

**zeppelin-config**
```
kubectl delete configmap zeppelin-config
kubectl create configmap zeppelin-config --from-file=zeppelin
```

- interpreter.json
- log4j.properties
- zeppelin-env.sh
- zeppelin-site.xml

**airflow-config**
```
kubectl delete configmap airflow-config
kubectl create configmap airflow-config --from-file=airflow
```

- airflow.cfg

**metastore-config**
`kubectl create configmap metastore-config --from-file=metastore`

- beeline-log4j2.properties.template
- hive-default.xml.template
- hive-env.sh.template
- hive-exec-log4j2.properties.template
- hive-log4j2.properties.template
- hive-site.xml
- ivysettings.xml
- llap-cli-log4j2.properties.template
- llap-daemon-log4j2.properties.template
- parquet-logging.properties

File: prediction.ml/python/Dockerfile [hunk 1]
```diff
@@ -2,26 +2,10 @@ FROM fluxcapacitor/package-java-openjdk-1.8:master
 
 WORKDIR /root
 
-RUN \
-  echo "deb https://dl.bintray.com/sbt/debian /" | tee -a /etc/apt/sources.list.d/sbt.list \
-  && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 2EE0EA64E40A89B84B2DF73499E82A75642AC823 \
```
Comment (BrentDorsey) [inline prediction.ml/python/Dockerfile]: removed duplicate code, keyserver and maven are already installed in package-java-openjdk-1.8
package-java-openjdk-1.8

- [keyserver](https://github.com/fluxcapacitor/pipeline/blob/master/package.ml/java/openjdk/1.8/Dockerfile#L13)
- [apache-maven-3.3.9](https://github.com/fluxcapacitor/pipeline/blob/master/package.ml/java/openjdk/1.8/Dockerfile#L21)

File: scheduler.ml/airflow/Dockerfile [hunk 2]
```diff
@@ -50,18 +51,16 @@ RUN \
   && cd $AIRFLOW_HOME
 
 RUN \
-  pip install airflow==1.7.1.3 \
+  pip install airflow==1.8.0 \
```
Comment (BrentDorsey) [inline scheduler.ml/airflow/Dockerfile]: updating to airflow latest "[1.8.0+apache incubating](https://github.com/apache/incubator-airflow/releases)" version

File: scheduler.ml/airflow/Dockerfile [hunk 3]
```diff
@@ -50,18 +51,16 @@ RUN \
...
   && pip install celery[redis] \
   && pip install flower
 
-RUN \
-  pip install tensorflow==1.0.1
```
Comment (BrentDorsey) [inline scheduler.ml/airflow/Dockerfile]: removing duplicate code, tensorflow is installed [here in package-spark-2.1.0](https://github.com/fluxcapacitor/pipeline/blob/master/package.ml/spark/2.1.0/Dockerfile#L84)

File: zeppelin.ml/Dockerfile [hunk 4]
```diff
@@ -24,16 +27,11 @@ RUN \
   && mv kubectl /usr/local/bin/kubectl
 
 # lib & notesbooks provided via Kubernets PersistentVolume Mount
-# from the github https://github.com/fluxcapacitor/source.ml repo 
+# from the github source.ml repo -> /root/pipeline
 # .kube/config is volume mounted from Kubernetes secret
 
-ENV \
-  TENSORFLOW_VERSION=1.0.1
```
Comment (BrentDorsey) [inline zeppelin.ml/Dockerfile]: removing duplicate code, tensorflow is installed [here in package-spark-2.1.0](https://github.com/fluxcapacitor/pipeline/blob/master/package.ml/spark/2.1.0/Dockerfile#L84)

**Comment by schnee** [comment]
This concept (externalizing config) seems like a great add to Pipeline.

**Comment by elgalu** [comment]
This quite a big PR and quite old and has conflicts, are you planning to close or re-work it?

**Pull request closed without merging**